source/core_sntp_client.c

```diff
@@ -339,6 +390,12 @@ SntpStatus_t Sntp_SendTimeRequest( SntpContext_t * pContext,
         LogError( ( "Invalid context parameter: Context cannot be NULL" ) );
     }
 
+    else if( isContextInitialized( pContext ) == false )
+    {
+        status = SntpErrorContextNotInitialized;
+        LogError( ( "Invalid parameter: Context is not initialized with Sntp_Init" ) );
```

Comment:
Is it possible to call Sntp_Init with arguments that end up making an invalid context?

Reply:
Nope, the point of this PR is to ensure that the Send/Receive APIs are not called without `Sntp_Init` having been called to initialize the context. Thus, the logic of `isContextInitialized` private function checks that the context represents data that would been initialized by the `Sntp_Init` function